### PR TITLE
Facility from subdomain

### PR DESCRIPTION
**Pull request by bglusman**
@fcapovilla / @komizutama / @net we could merge this as is but may want to build some new index pages for finding all facilities and/or scope /manage/stocks to the current facility etc, but this works locally provided you add the subdomains used to /etc/hosts, and defaults to same first facility in most cases as it does at present.

Thanks to @net for starting this off a couple months ago, we're planning to start hosting the app at openpantry.io soon and using subdomains there, plus perhaps hosting an informational page at the app root/www.  Since there's so much focus on food and pantries/soup kitchens around Thanksgiving, we're hoping maybe we can make a push to get a usable v1 in front of some extra users around/before Thanksgiving, or perhaps between Thanksgiving and Christmas.  You've done a bunch of amazing and valuable work recently @fcapovilla , if you think you have some additional time available in that timeframe it may help us meet some milestones!  I'm hoping I'll have more time the next couple weeks also.  Things are still rudimentary, but hopefully we can prioritize next steps and get this in front of some real users to figure out how far we have to go.  Maybe next step from here is having a basic-auth password per subdomain, instead of globally, so a couple facilities can manage themselves without stepping on each other?  Not sure, @komizutama looking at you here.

### CODE REVIEW

**Comment by fcapovilla** [comment]
I only have some free time on the week-ends, so I'll try to tackles some issues then.

Is there a milestone on github when I can find all the high priority issues?
I've found the "MVP" milestone which still has some open issues, but this milestones seems to be closed?

**Comment by bglusman** [comment]
@fcapovilla of course, and don't want to take all your time/ask too much of you!  Just trying to plan out what we can get done, and what's highest priority to make it testable/usable by multiple org's at once.. until now it's just been Masbia in a single facility testing occasionally, but we need some new stakeholders to validate what we have is useful and practical for them.  If you have any contact with org's in Montreal, or are willing to reach out and/or help train them in using it a little bit, that might be invaluable, and @komizutama and I will be looking for more similar opportunities around NYC.   Any playing around you do that finds usability issues or ideas you have are also super welcome, we don't have a lot of special expertise here, though @komizutama has been working at and with Masbia for about a year now, it's likely that a lot of other pantries operate differently and we'll need to widen perspective beyond their experience at Masbia soon.  If you want to take a crack at writing ideas for issues of your own, please feel free!  I can try and write something up for the idea of extending auth to be facility specific (for now an org with multiple facilities will probably just manage each facility seperatley, maybe with one user/login having permission to arbitrary number of facilities).  It doesn't have to be basic_auth of course, we can extend current user model to have real login, but we also need to maintain ability for people without email addresses or phones to get logged in by a facility admin, since of course much of the population served is older, not online, and/or homeless, but we don't have anything set in stone for best way to achieve this.  If you wanted to integrate coherence or uberauth alongside the current Guardian based login, that might be really useful.  I have some vague notions how this might work, but was putting that off until needed.  Basically, feel empowered to step up from the amazing contributions you've made so far and consider yourself a co-maintainer and help set direction for the project wherever you see the opportunity!  @komizutama and Masbia had the original vision for this software, but it's taken a lot of just digging in and taking a crack at our best ideas since then to get where we are, and we're getting close, but still have a ways to go :-)

**Review by fcapovilla — APPROVED** [review]
The code looks good. Nice use of Plugs.
I did some tests in local and it seems to work as intended, but only for the UserSelection controller.

When testing, I encountered this error when accessing the usr_selections route, but it seems unrelated with your branch. Might be a problem with my local instance : 
** (FunctionClauseError) no function clause matching in Guardian.jose_jwk/1

**Comment by bglusman** [comment]
@fcapovilla oh interesting, for me it worked on food_selections also, but I'll see if I can repro that.  I also attempted to capture a first draft at the new epic I referred to above to make multi-facility support more or less "real".  Not sure if we'll merge this in ahead of that, or turn this branch into an epic/feature branch for all of that related work.  Also need @komizutama to weigh in on this, and agree on priority, but your reactions and interest/relevant experience there is welcome!

**Comment by fcapovilla** [comment]
Looking at the FoodSelectionController code, it seems the controller uses the current user's facility and not the subdomains's facility.
Since my UserSelection controller doesn't work, no user is selected and I'm always in guest mode. My guest user is assigned to the first facility, so I always get the first facility on all subdomains.

**Comment by bglusman** [comment]
ahh, yes, we should probably also redirect to the current user's facility when user is logged in and NOT on their facility, and look for that part before setting facility from subdomain I guess... or something.... may have to play around with that, though if you go to /user_selections path on an existing facility subdomain it should allow you to see or create users in that facility, and then login as one that will be in correct facility.

**Comment by bglusman** [comment]
oh also, may need to reseed your database or manually add subdomains since the migration (I assume you ran) adds the fields but won't add any specific subdomains unless seeding.

**Comment by fcapovilla** [comment]
I know, I ran the migration and added the subdomains ;) . I can confirm the user_selections route does the correct SQL queries and uses the correct facility on each subdomain.

I just can't create or select users there because of the Guardian errors.
I'll try to find the cause of my errors this week-end.

**Comment by bglusman** [comment]
Oh!  Did you add the env variable for Guardian? You need to export GUARDIAN_SECRET_KEY to app, and make sure it's there at app launch.  Should make it so there's some default value there in Dev, and that's only needed in Prod... apologies...

**Comment by fcapovilla** [comment]
To answer you other message, I do not have any contact with orgs in Montreal or Quebec. I'm more interested in contributing code, features and translations to the project since communication is not my forte.

Since I'm not entirely familiar with the user's needs and requirements on this project, having a clear milestone would be really helpful so I can focus on high-priority issues in order to get to a stable v1 as fast as possible.

I'll also try to remember to create issues if I have an idea for new features or improvements.

Thanks you!

**Comment by fcapovilla** [comment]
Exporting GUARDIAN_SECRET_KEY worked. I was able to create a user, and access the food_selections page with this user.
I can confirm the facilities work correctly when a user is selected.

**Comment by bglusman** [comment]
Awesome, made a new issue to make that more ergonomic, I'd meant to initially and keep forgetting how many people it's messed up!  #168

**Comment by bglusman** [comment]
Hopefully #167 and the individual issues in there are reasonably well scoped maybe you and I can tackle them together over next week or two, dunno...  but for now, thanks again and hope you're having a good night!

**Comment by komizutama** [comment]
Hi Sorry. I got wrangled into getting donations we acquired for Puerto Rico out the door and have been driving a van around Brooklyn and whatnot today, woo. @fcapovilla I'll work on writing some better issues, grooming, and generating a product roadmap/milestone and get it into the issues. As Brian said, if you're interested in being a co-maintainer I'm definitely down. I know we could use some more dedicated programmers... I've been mostly focussed on trying to Scope/PM/Design this when I can get time. I'll work on generating a clearer picture over the next few days. I have done some mock-up cleanup, but I think that I can get better and also get images into the actual issues as well as write things more clearly. 
I do think having multi-agency may work best because it can serve for both multiple sites (a Masbia Issue) and multiple orgs (Where we as OP want it to move). 
I'm definitely down to have more co-maintainers on board with this project. Would love to have a conversation with you sometime soon, just to get to know you a bit better since I haven't interacted with you as much as @bglusman has.

**Comment by fcapovilla** [comment]
@komizutama : I received an invitation to Slack, but slack is really slow right now so I'll answer here.

Currently, I can only use a little of my free time to contribute to your project because I already have a 40-hour work week to do, so I think it's too early to make me a co-maintainer.
Still, I'm happy to use my skills to help your project grow.

**Comment by bglusman** [comment]
@fcapovilla no worries, don't feel pressured, just welcomed and appreciated!  No hard press, just excited you've been doing so much and wanted to recognize and encourage/welcome you to the team!

@komizutama Can you play around with this branch locally/maybe try out a rebase/merge of it with your color tabs branch and make sure no concerns before we merge in?  It's def not perfect, but probably good enough to move ahead, and maybe push up to staging and play around with making sure we know how to configure DNS there.. I think I have a spare domain we can play around with as a multi-facility staging domain, or you may have one registered alongside openpantry.io so we can ensure setup is identical/easy to reproduce success there to production and debug any future issues side by side, or can even buy a new domain for $10 like openpantrystaging.org or something if want.

**Comment by komizutama** [comment]
Yeah, I will. I'm kinda of in the middle of something else though. When I get this finished i'll swap over.

**Comment by komizutama** [comment]
This all looks good. Thank you @net and @bglusman!

**Pull request merged**
